src/code_e92d0_len_5da0.c

```diff
@@ -120,79 +235,173 @@ INCLUDE_ASM(code_e92d0_len_5da0, si_handle_get_4_float);
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_set_priority);
+s32 si_handle_set_priority(script_context* script) {
+    set_script_priority(script, get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_set_timescale);
+s32 si_handle_set_timescale(script_context* script) {
+    set_script_timescale(script, get_float_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_set_group);
+s32 si_handle_set_group(script_context* script) {
+    set_script_group(script, get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_suspend_all);
+s32 si_handle_suspend_all(script_context* script) {
+    suspend_all_group(get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_resume_all);
+s32 si_handle_resume_all(script_context* script) {
+    resume_all_group(get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_suspend_others);
+s32 si_handle_suspend_others(script_context* script) {
+    suspend_group_others(script, get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_resume_others);
+s32 si_handle_resume_others(script_context* script) {
+    resume_group_others(script, get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_suspend);
+s32 si_handle_suspend(script_context* script) {
+    suspend_all_script(get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_resume);
+s32 si_handle_resume(script_context* script) {
+    resume_all_script(get_variable(script, *script->ptrReadPos));
+    return 2;
+}
 
 INCLUDE_ASM(code_e92d0_len_5da0, si_handle_does_script_exist);
 
 INCLUDE_ASM(code_e92d0_len_5da0, si_handle_bind_lock);
 
 INCLUDE_ASM(code_e92d0_len_5da0, si_handle_thread);
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_end_thread);
+s32 si_handle_end_thread(script_context* script) {
+    kill_script(script);
+    return 255;
+}
 
 INCLUDE_ASM(code_e92d0_len_5da0, si_handle_child_thread);
 
-INCLUDE_ASM(code_e92d0_len_5da0, si_handle_end_child_thread);
+s32 si_handle_end_child_thread(script_context* script) {
+    kill_script(script);
+    return 0;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, func_802C6E14);
+s32 func_802C6E14(script_context* script) {
+    return 2;
+}
 
 INCLUDE_ASM(code_e92d0_len_5da0, si_handle_print_debug_var);
 
-INCLUDE_ASM(code_e92d0_len_5da0, func_802C739C);
+s32 func_802C739C(script_context* script) {
+    script->ptrSavedPosition = *script->ptrReadPos;
+    return 2;
+}
+
+s32 func_802C73B0(script_context* script) {
+    return 2;
+}
 
-INCLUDE_ASM(code_e92d0_len_5da0, func_802C73B0);
+s32 func_802C73B8(script_context* script) {
+    s32 i;
 
-INCLUDE_ASM(code_e92d0_len_5da0, func_802C73B8);
+    for (i = 0; i < 0x80; i++) {
```

Comment:
do we know what this is the size of? If not, all good to leave as-is, though maybe a todo should be left.

Reply:
`ARRAY_COUNT(gCurrentScriptListPtr)`, given that each iteration accesses `(*gCurrentScriptListPtr)[index]`